mlflow/utils/proto_json_utils.py

```diff
@@ -99,14 +99,25 @@ def _dataframe_from_json(
     from mlflow.types import DataType
 
     if schema is not None:
-        dtypes = dict(zip(schema.input_names(), schema.pandas_types()))
+        if schema.is_tensor_spec():
+            # The schema can be either:
+            #  - a single tensor: attempt to parse all columns with the same dtype
+            #  - a dictionary of tensors: each column gets the type from equaly named tensor
+            if len(schema.inputs) == 1:
+                dtypes = schema.numpy_types()[0]
+            else:
+                dtypes = dict(zip(schema.input_names(), schema.numpy_types()))
+        else:
+            dtypes = dict(zip(schema.input_names(), schema.pandas_types()))
```

Comment:
should we check that input names isn't none first?

Reply:
discussed offline: the input names are actually returned as integers 0..N in case the schema is unnamed.